### PR TITLE
MM-12320 - Change "about" section references to use the site name when it is configured

### PR DESCRIPTION
**Pull request by reflog**
#### Summary
When Site Name is configured under "System Console > Custom Branding > Site Name:

update Main Menu from "About Mattermost" to "About [Site Name]"

update About modal heading "About Mattermost" to "About [Site Name]" (see attached image)

This should apply to desktop app, Webapp and RN. 

#### Ticket Link
Fixes https://mattermost.atlassian.net/browse/MM-12320

#### Related Pull Requests
- Has mobile changes (please link here)
https://github.com/mattermost/mattermost-mobile/pull/2835

### CODE REVIEW

File: components/about_build_modal/about_build_modal.test.jsx [hunk 1]
```diff
@@ -129,6 +128,6 @@ describe('components/AboutBuildModal', () => {
             ...props,
         };
 
-        return shallow(<AboutBuildModal {...allProps}/>);
+        return shallowWithIntl(<AboutBuildModal {...allProps}/>);
```
Comment (jespino) [inline components/about_build_modal/about_build_modal.test.jsx]: I don't see the need of the shallowWithIntl here, `AboutBuildModal` doesn't use intl in the shallow level.

File: components/admin_console/admin_navbar_dropdown/admin_navbar_dropdown.jsx [hunk 2]
```diff
@@ -20,9 +20,11 @@ import MenuItemExternalLink from 'components/widgets/menu/menu_items/menu_item_e
 import MenuItemToggleModalRedux from 'components/widgets/menu/menu_items/menu_item_toggle_modal_redux';
 import MenuItemBlockableLink from 'components/widgets/menu/menu_items/menu_item_blockable_link';
 
-export default class AdminNavbarDropdown extends React.Component {
+class AdminNavbarDropdown extends React.Component {
     static propTypes = {
         locale: PropTypes.string.isRequired,
+        siteName: PropTypes.string,
+        intl: intlShape.isRequired,
```
Comment (jespino) [inline components/admin_console/admin_navbar_dropdown/admin_navbar_dropdown.jsx]: We no longer use this approach, we now use the approach that you can see here `components/select_team/components/select_team_item.jsx`. There is a Context provider for intl.

File: components/main_menu/main_menu.jsx [hunk 3]
```diff
@@ -33,13 +34,15 @@ import MenuItemLink from 'components/widgets/menu/menu_items/menu_item_link.jsx'
...
-export default class MainMenu extends React.PureComponent {
+export class MainMenu extends React.PureComponent {
     static propTypes = {
         mobile: PropTypes.bool.isRequired,
         id: PropTypes.string,
         teamId: PropTypes.string,
         teamType: PropTypes.string,
         teamName: PropTypes.string,
+        siteName: PropTypes.string,
+        intl: intlShape.isRequired,
```
Comment (jespino) [inline components/main_menu/main_menu.jsx]: same here

File: components/about_build_modal/about_build_modal.test.jsx (hunk 1 above)
Comment (reflog) [reply components/about_build_modal/about_build_modal.test.jsx]: FormattedMessage requires it. If you use regular `shallow` it errors with:

```
 console.error tests/setup.js:55
    [React Intl] Could not find required `intl` object. <IntlProvider> needs to exist in the component ancestry. Using default message as fallback.

  console.error tests/setup.js:55
    The above error occurred in the <InjectIntl(FormattedMarkdownMessage)> component:
        in InjectIntl(FormattedMarkdownMessage) (created by AboutBuildModal)
```

File: components/main_menu/main_menu.test.jsx [hunk 4]
```diff
@@ -3,10 +3,21 @@
...
 import React from 'react';
 import {shallow} from 'enzyme';
+import {IntlProvider} from 'react-intl';
 
 import {Constants} from 'utils/constants.jsx';
 
-import MainMenu from './main_menu.jsx';
+import {MainMenu} from './main_menu.jsx';
+
+function shallowWithIntl(Component, props) {
```
Comment (jespino) [inline components/main_menu/main_menu.test.jsx]: Using the new approach with the context, you can directly use the `shallowWithIntl` function from the `tests/helpers/intl-test-helper.jsx` module.

File: components/about_build_modal/about_build_modal.test.jsx (hunk 1 above)
Comment (jespino) [reply components/about_build_modal/about_build_modal.test.jsx]: It doesn't happen in my machine. Works perfectly fine with shallow. Actually the error that you are pasting there is not related to the change, is about the `FormattedMarkdownMessage` so it shouldn't be related to this.

Comment (reflog) [reply components/about_build_modal/about_build_modal.test.jsx]: 👍

File: components/about_build_modal/about_build_modal.jsx [hunk 5]
```diff
@@ -211,7 +211,8 @@ export default class AboutBuildModal extends React.PureComponent {
                     >
                         <FormattedMessage
                             id='about.title'
-                            defaultMessage='About Mattermost'
+                            values={{appTitle: config.SiteName || 'Mattermost'}}
```
Comment (enahum) [inline components/about_build_modal/about_build_modal.jsx]: just to point out this is the same explanation as in the mobile PR, 0/5 if you change it or not

File: components/admin_console/admin_navbar_dropdown/index.js [hunk 6]
```diff
@@ -16,6 +16,7 @@ function mapStateToProps(state) {
     return {
         locale: getCurrentLocale(state),
         teams: getMyTeams(state),
+        siteName: state.config.SiteName,
```
Comment (mkraft) [inline components/admin_console/admin_navbar_dropdown/index.js]: @reflog This is crashing the app on master when one clicks the hamburger menu from the system console.

```suggestion
        siteName: state.entities.general.config.SiteName,
```

Comment (reflog) [reply components/admin_console/admin_navbar_dropdown/index.js]: I know :( Already submitted a PR https://github.com/mattermost/mattermost-webapp/pull/2877

**Pull request merged**